CMakeLists.txt

```diff
@@ -5,19 +5,30 @@ PROJECT(qmcpack)
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8.10)
 IF(COMMAND cmake_policy)
   cmake_policy(SET CMP0003 NEW)
+  IF(NOT($ENV{CRAYPE_VERSION} MATCHES "."))
+    MESSAGE(STATUS "Running on a Cray machine.")
+    cmake_policy(SET CMP0056 NEW)
+    # This policy insures that the CMAKE_EXE_LINKER_FLAGS of the calling project
+    # are used in the try_compile test cmake project.  Without this
+    # clean rpath builds with mkl aren't possible. While it is unknown by
+    # cmake 2.8.10.2 on titan it still breaks the iszero test there --PWD
```

Comment:
I think the warning is something you want to see.  If you have a cmake that old the iszero and c++11 test can fail if you are trying to use a compiler without polluting the Library Paths.  Unfortunately somehow on cray with cmake 2.8.10 you get the warning and a failure of the iszero test.

Reply:
The test should use the CMake version, and not whether it's a Cray system.

It's okay to have degraded functionality if building with older versions of CMake.   (In this case, the ability to use rpath consistently. )   Seeing the CMake warning about CMP0056 is not very helpful - it would be better to print a more user-friendly warning message.

The iszero test is a compile-only test - runtime library paths should not enter into it.

Reply:
it does now

Reply:
If the linker flags are not set correctly this fails because -fopenmp and other flags are added but they can't be linked.  So you don't just get a test failure you get a cmake fail.

Reply:
Do you have an example on how to reproduce this failure (#685 is light on details)?

The iszero function only appears in fairly new glibc versions (>=2.25).  I would expect the test to even fail to compile on most systems.  And even if the system has it, and it fails to link, it should still not cause the cmake configuration step to fail.  The failure mode will be a 'redefinition of iszero' error during compilation. See #429 for more about the iszero test.

The c++11 flag test seems more likely to be a problem, where link flag problems will lead to the CMake configuration file stopping.

Reply:
Updated #685, I should say the iszero fails, but it also halts configuration.